flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerTest.java

```diff
@@ -544,21 +544,17 @@ public void testDuplicatePendingSlotRequestAfterSuccessfulAllocation() throws Ex
 	@Test
 	public void testAcceptingDuplicateSlotRequestAfterAllocationRelease() throws Exception {
 		final ResourceManagerId resourceManagerId = ResourceManagerId.generate();
-		final ResourceActions resourceManagerActions = mock(ResourceActions.class);
+		final AtomicInteger allocateResourceCalls = new AtomicInteger(0);
+		final ResourceActions resourceManagerActions = new TestingResourceActionsBuilder()
+			.setAllocateResourceConsumer(resourceProfile -> allocateResourceCalls.incrementAndGet())
```

Comment:
the logic of some of this built TestingResourceActions repeat in some places, we could create some standard factory methods for them (maybe in future), e.g.:
```
resourceManagerActions = TestingResourceActionsBuilder.createAllocatingFixedSlotsResource(numOfSlotsPerWorker);
allocateResourceCalls = resourceManagerActions.getAllocationsCounter();
```

Reply:
True, one could add a helper function doing this.